Screens/LoginScreen.js

```diff
@@ -1,93 +1,116 @@
-// LoginScreen.js
 import React, { useState } from "react";
-import {
-  TouchableOpacity,
-  View,
-  Text,
-  TextInput,
-  Button,
-  StyleSheet,
-} from "react-native";
+import { TouchableOpacity, StyleSheet, View } from "react-native";
+import { Text } from "react-native-paper";
+import Background from "../components/Background";
+import Logo from "../components/Logo";
+import Header from "../components/Header";
+import Button from "../components/Button";
+import TextInput from "../components/TextInput";
+import BackButton from "../components/BackButton";
+import { theme } from "../core/theme";
+import { usernameValidator } from "../utils/UsernameValidator";
+import { passwordValidator } from "../utils/PasswordValidator";
 
 import axios from "axios";
 import AsyncStorage from "@react-native-async-storage/async-storage";
 import RestApiService from "../services/RestApiService";
 
 const LoginScreen = ({ navigation }) => {
-  const [username, setUsername] = useState("");
-  const [password, setPassword] = useState("");
+  const [username, setUsername] = useState({ value: "", error: "" });
+  const [password, setPassword] = useState({ value: "", error: "" });
 
   const handleLogin = async () => {
     // Implement your login logic here
     console.log("Username:", username);
     console.log("Password:", password);
     // For a real application, you would send the credentials to your server for authentication
+
+    const usernameError = usernameValidator(username.value);
+    const passwordError = passwordValidator(password.value);
+
+    if (usernameError || passwordError) {
+      setUsername({ ...username, error: usernameError });
+      setPassword({ ...password, error: passwordError });
+      return;
+    }
+
     try {
-      console.log("before")
+      console.log("before");
```

Comment:
If it is possible we want to have a clean logs in main, therefore we should remove any unnecessary logs